enhancements/update/openstack-cacerts.md

```diff
@@ -0,0 +1,263 @@
+[source-ciro]: https://github.com/openshift/cluster-image-registry-operator/blob/f27308c19e654021b614b12505b298315a72135c/pkg/storage/swift/swift.go#L176-L189
+[source-cncc]: https://github.com/openshift/cloud-network-config-controller/blob/5985f3b31274c555b32ac59aa0dfcbca877a3235/cmd/cloud-network-config-controller/main.go#L200-L224
+[source-cco]: https://github.com/openshift/cluster-config-operator/blob/cbfd00a8204d5ae5d755edf05b7c26a60218a9b7/pkg/operator/kube_cloud_config/controller.go#L113-L132
+
+In [OSASINFRA-3729](https://issues.redhat.com/browse/OSASINFRA-3729), we have
+proposed resolving these issues by making the CA cert part of the credential
+secret managed by Cloud Credential Operator (CCO). CCO provides the ability to
+copy a set of credentials from a well-known location (`kube-system /
+openstack-credentials` for OpenStack) to a namespace by creating a
+`CredentialsRequest`. CCO previously managed a single field, `clouds.yaml`,
+which contained OpenStack credentials in the typical format (note that there is
+also an additional field, `clouds.conf`, which was used to configure the legacy
+in-tree CCM and is still being added by the Installer, but CCO ignores this).
+With this change, CCO now manages a new `cacert` key, copying this to the
+credentials secrets created in response to `CredentialsRequest` CRs. This
+allows us to simplify a lot of operators by removing the config map syncers
+they currently have as well as removing a split between standalone and
+hypershift deployments due to differences in config map naming. However,
+because of CCM's role in bootstrapping in OpenShift, we cannot rely on
+CCO-managed credentials (tl;dr: CCM comes up before CCO does and needs to talk
+to the cloud so that it can get instances to deploy pods on). This means we
+cannot remove the CA cert from the CCM-specific config map. However, we do not
+want CA cert rotation to involve updating multiple locations..
+
+The dumb solution is to introduce a controller that will either copy the CA
+cert from the credential secret to the CCM config map or vice versa. However,
+this poses an issue for tooling. If we copy from the credential secret to the
+CCM config map, we break hypothetical tooling that is updating the config map,
+since all changes to this config map will be overridden. If we copy from the
+CCM config map to the credential secret, we prevent new tooling being written
+that targets the credential secret since, again, any changes will be
+overwritten. We need to find a solution that allow existing tooling to continue
+to work for some transition period, while also allowing tooling that uses the
+new location to be written.
+
+### User Stories
+
+* As an OpenShift administrator, I want to be able to rotate the CA cert used
+  for my cloud as easily as I can rotate my credentials.
```

Comment:
I think this use case is correct, but I don't think it necessarily follows that the user expects the CA cert and cloud credentials to be in the same place. We want them to be managed from only one place each. Purely from the user's POV, I struggle to convince myself that it's important that they're both in the same place.

I see rotating a CA cert and rotating credentials as being quite different operations with quite different lifecycles. Do we have any evidence that users regularly do them at the same time?